### PR TITLE
Update "How to deploy LLM" blog post to use `huggingface_hub` in example 

### PR DESCRIPTION
**Pull request by Wauplin**
Related to https://github.com/huggingface/blog/pull/1286#discussion_r1252786764.

Goal is to showcase `huggingface_hub.InferenceClient` in order to get more visibility/usage on a single client library. Implementation is (almost) the same as `text-generation` cc @OlivierDehaene @julien-c @philschmid 

`InferenceClient` has been released in `huggingface_hub==0.16` today (see [release notes](https://github.com/huggingface/huggingface_hub/releases/tag/v0.16.2)). I tested the modification in this PR by creating a temporary Inference Endpoint.

---

Worth mentioning that we can have a simpler example here. If `details=False` is passed (default), tokens are yielded as `str` directly. I still think that showing how to use `details=True` in the blog post is better.

```py
stream = client.text_generation(prompt, stream=True, **gen_kwargs)

# yield each generated token
for token in stream:

    # yield the generated token
    print(token, end="")
    # yield token
```

### CODE REVIEW

**Review by julien-c — APPROVED** [review]
i think it makes sense!

maybe still mention `pip install text-generation` as another option but i don't think it's wort it

**Comment by osanseviero** [comment]
> maybe still mention pip install text-generation as another option but i don't think it's wort it

I would keep it simple with a single option

File: inference-endpoints-llm.md [hunk 1]
```diff
@@ -89,23 +89,23 @@ Requesting and generating text with LLMs can be a time-consuming and iterative p
...
-First, you need to install the `text-generation` client
+First, you need to install the `huggingface_hub` library:
 
 ```python
-pip install text-generation
+pip install -U huggingface_hub
 ```
 
-We can create a `Client` providing our endpoint URL and credential alongside the hyperparameter we want to use
+We can create a `InferenceClient` providing our endpoint URL and credential alongside the hyperparameter we want to use
```
Comment (pcuenca) [inline inference-endpoints-llm.md:98]:
```suggestion
We can create a `InferenceClient` providing our endpoint URL and credential alongside the hyperparameters we want to use
```

**Review by pcuenca — APPROVED** [review]

**Comment by Wauplin** [comment]
Thanks for the quick reviews. I merging this :)

**Pull request merged**